lib/snipr/process_signaller.rb

```diff
@@ -132,14 +132,23 @@ def dry_run
       @dry_run = true
     end
 
+    ##
+    # Ensure that the parent PID is still accurate
+    def ppid_matches?(process)
+      # PPID is the 4th item under /proc/pid/stat (whitespace-delimited)
+      process.ppid.to_i == File.read("/proc/#{process.pid}/stat").split[3].to_i
```

Comment:
Prefer `Integer()` to `String#to_i`, the latter is too permissive -- any unrecognized string becomes `0`.